src/data/mv_polynomial.lean

```diff
@@ -108,13 +108,15 @@ variables [comm_semiring α] {p q : mv_polynomial σ α}
 
 instance decidable_eq_mv_polynomial [decidable_eq σ] [decidable_eq α] :
   decidable_eq (mv_polynomial σ α) := finsupp.decidable_eq
-instance : has_zero (mv_polynomial σ α) := finsupp.has_zero
-instance : has_one (mv_polynomial σ α) := finsupp.has_one
-instance : has_add (mv_polynomial σ α) := finsupp.has_add
-instance : has_mul (mv_polynomial σ α) := finsupp.has_mul
-instance : comm_semiring (mv_polynomial σ α) := finsupp.comm_semiring
+instance : comm_semiring (mv_polynomial σ α) := add_monoid_algebra.comm_semiring
 instance : inhabited (mv_polynomial σ α) := ⟨0⟩
 
+/-- the coercion turning an `mv_polynomial` into the function which reports the coefficient of a given monomial -/
```

Comment:
I wouldn't mind seeing the last of this `coe`. It's a very unmathematical notation.

Reply:
I guess we can repair most of the breakage locations right away with `mv_polynomial.coeff`. But maybe not in this PR.

Reply:
Noted, but I'm not going to attempt this here.